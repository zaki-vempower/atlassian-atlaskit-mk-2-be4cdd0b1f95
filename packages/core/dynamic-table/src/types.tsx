import React from 'react';
import {
  UIAnalyticsEvent,
  WithAnalyticsEventsProps,
} from '@atlaskit/analytics-next';

export interface RowCellType {
  key?: string | number;
  colSpan?: number;
  content?: React.ReactNode | string;
  testId?: string;
}

export interface I18nShape {
  prev: string;
  next: string;
}

export interface StatelessProps extends WithAnalyticsEventsProps {
  caption?: React.ReactNode;
  /** Object describing the column headings */
  head?: HeadType;
  /** The data to render in the table */
  rows?: Array<RowType>;
  /** Component to render when there is no data */
  emptyView?: React.ReactElement<any>;
  /** Controls the size of the rendered spinner */
  loadingSpinnerSize?: LoadingSpinnerSizeType;
  /** Whether to show the loading state or not */
  isLoading?: boolean;
  isFixedSize?: boolean;
  /** The maximum number of rows per page. No maximum by default. */
  rowsPerPage?: number;
  /** Called when the page changes. Provides an analytics event when the page change was from a click on pagination component. */
  onSetPage?: (page: number, UIAnalyticsEvent?: UIAnalyticsEvent) => void;
  /** Called when a column is sorted. Provides information about what was sorted and an analytics event. */
  onSort?: (data: any, UIAnalyticsEvent?: UIAnalyticsEvent) => void;
  /** Called after body table render when visible table rows change with table rows given. */
  onPageRowsUpdate?: (pageRows: Array<RowType>) => void;
  /** The current page number */
  page?: number;
  defaultPage?: number;
  /** The property to sort items by */
  sortKey?: string;
  /** Whether to sort in ascending or descending order */
  sortOrder?: SortOrderType;
  /** Whether to allow 'rank' sort in addition to ascending/descending */
  isRankable?: boolean;
  isRankingDisabled?: boolean;
  onRankStart?: (rankStart: RankStart) => void;
  onRankEnd?: (rankEnd: RankEnd, uiAnalyticsEvent?: UIAnalyticsEvent) => void;
  paginationi18n?: I18nShape;
  /** It highlights the passed row number on the current visible page. Starts with 0 */
  highlightedRowIndex?: number;
  /**
   * A `testId` prop is provided for specified elements, which is a unique string that appears as a data attribute
   * `data-testid` in the rendered code, serving as a hook for automated tests.
   *
   * The value of `testId` is used to prefix testId props in given elements:
   * **Notes: The content of the cell, keys, index are defined by consumers.
   * `${testId}--table` - Table.
   * `${testId}--head` - Table header.
   * `${testId}--head--{content of the cell}` - Table header cell can be identified by their content.
   * `${testId}--row--{index - content of the first cell}` - Table row.
   * `${testId}--body` - Table body.
   * `${testId}--body--{content of the cell}` - Table body cell can be identified by their content.
   **/
  testId?: string;
}

export interface StatefulProps extends WithAnalyticsEventsProps {
  caption?: Node | string;
  head?: HeadType;
  rows?: Array<RowType>;
  emptyView?: React.ReactElement<any>;
  loadingSpinnerSize?: LoadingSpinnerSizeType;
  isLoading?: boolean;
  isFixedSize?: boolean;
  rowsPerPage?: number;
  onSetPage?: (page: number, UIAnalyticsEvent?: UIAnalyticsEvent) => void;
  onSort?: (data: any, UIAnalyticsEvent?: UIAnalyticsEvent) => void;
  onPageRowsUpdate?: (pageRows: Array<RowType>) => void;
  page?: number;
  defaultPage?: number;
  sortKey?: string;
  defaultSortKey?: string;
  sortOrder?: SortOrderType;
  defaultSortOrder?: SortOrderType;
  isRankable?: boolean;
  isRankingDisabled?: boolean;
  onRankStart?: (rankStart: RankStart) => void;
  onRankEnd?: (rankEnd: RankEnd) => void;
  paginationi18n?: I18nShape;
  highlightedRowIndex?: number;
  testId?: string;
}

export type RowType = {
  cells: Array<RowCellType>;
  key?: string;
  onClick?: React.MouseEventHandler;
  testId?: string;
};

export type SortOrderType = 'ASC' | 'DESC';

export type SpinnerSizeType =
  | 'xsmall'
  | 'small'
  | 'medium'
  | 'large'
  | 'xlarge';

export type LoadingSpinnerSizeType = 'small' | 'large';

export interface HeadCellType extends RowCellType {
  isSortable?: boolean;
  width?: number;
  shouldTruncate?: boolean;
}

export interface RankEndLocation {
  index: number; // index on current page
  afterKey?: string;
  beforeKey?: string;
}

export interface RankEnd {
  sourceIndex: number;
  sourceKey: string;
  destination?: RankEndLocation;
}

export interface RankStart {
  index: number;
  key: string;
}

export interface HeadType {
  cells: Array<HeadCellType>;
}
