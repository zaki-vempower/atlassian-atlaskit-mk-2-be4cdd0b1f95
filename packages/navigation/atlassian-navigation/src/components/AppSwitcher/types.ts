import { IconButtonProps } from '../IconButton/types';

export type AppSwitcherProps = Omit<IconButtonProps, 'icon' | 'tooltip'> & {
  tooltip: React.ReactNode;
};
