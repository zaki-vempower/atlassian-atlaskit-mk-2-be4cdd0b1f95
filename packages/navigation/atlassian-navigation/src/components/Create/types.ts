export type CreateProps = {
  text: string;
  buttonTooltip?: React.ReactNode;
  iconButtonTooltip?: React.ReactNode;
  onClick: (e: React.MouseEvent<HTMLElement>) => void;
};
