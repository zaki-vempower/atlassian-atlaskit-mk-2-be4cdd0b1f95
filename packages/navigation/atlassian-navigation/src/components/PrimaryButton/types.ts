import { ButtonProps } from '@atlaskit/button/types';

export type PrimaryButtonProps = Omit<ButtonProps, 'appearance'> & {
  /** Prop used to highlight the button is it is the active item */
  isHighlighted?: boolean;
  /**  */
  testId?: string;
  /** A string or React Node to render as a tooltip */
  tooltip?: React.ReactNode;
};

export type PrimaryButtonSkeletonProps = {
  className?: string;
};
