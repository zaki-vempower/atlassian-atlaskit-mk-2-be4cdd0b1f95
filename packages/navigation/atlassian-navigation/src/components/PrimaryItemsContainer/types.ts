import { ComponentType, ReactNodeArray, ReactNode } from 'react';

export type PrimaryItemsContainerProps = {
  moreLabel: ReactNode;
  items: ReactNodeArray;
  create?: ComponentType<{}>;
};

export type PrimaryItemsContainerSkeletonProps = {
  count: number;
};
