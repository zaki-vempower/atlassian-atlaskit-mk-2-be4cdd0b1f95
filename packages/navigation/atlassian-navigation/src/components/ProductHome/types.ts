import { ComponentType, MouseEvent } from 'react';
import { LogoProps } from '@atlaskit/logo';

export type ProductHomeProps = {
  /** The product icon. Expected to be an Icon from the Atlaskit Logo package. Visible on smaller screen sizes */
  icon: ComponentType<Partial<LogoProps>>;
  /** The product logo, visible on larger screen sizes */
  logo: ComponentType<Partial<LogoProps>>;
  /** Optional onClick handler */
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  /** Optional mouseDown handler */
  onMouseDown?: (event: MouseEvent<HTMLElement>) => void;
  /** Href to be passed to product home */
  href?: string;
  /** Name of the site that appears next to the logo **/
  siteTitle?: string;
};

export type CustomProductHomeProps = {
  iconAlt: string;
  iconUrl: string;
  logoAlt: string;
  logoUrl: string;
  /** Optional onClick handler */
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  /** Optional mouseDown handler */
  onMouseDown?: (event: MouseEvent<HTMLElement>) => void;
  /** Href to be passed to product home */
  href?: string;
  /** Name of the site that appears next to the logo **/
  siteTitle?: string;
};
