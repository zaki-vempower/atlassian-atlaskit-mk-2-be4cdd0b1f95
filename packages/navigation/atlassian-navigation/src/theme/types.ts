import { CSSObject } from '@emotion/core';

export type CSSProperties = CSSObject & {
  backgroundColor: string;
  color: string;
};

export type ButtonCSSProperties = CSSProperties & {
  boxShadow: string;
};

export type ButtonFocusCSSProperties = {
  boxShadow: string;
  color: string;
  backgroundColor: string;
};

export type ButtonCSSContext = {
  default: ButtonCSSProperties;
  hover: ButtonCSSProperties;
  focus: ButtonFocusCSSProperties;
  active: ButtonCSSProperties;
  selected: ButtonCSSProperties;
};

export type CreateCSS = ButtonCSSContext;

export type IconButtonCSS = ButtonCSSContext;

export type PrimaryButtonCSS = ButtonCSSContext;

export type NavigationCSS = CSSProperties;

type LogoStyleProps = {
  iconGradientStart?: string;
  iconGradientStop?: string;
  iconColor?: string;
  textColor?: string;
};
export type ProductHomeCSS = CSSProperties & LogoStyleProps;

export type SearchCSS = {
  default: CSSProperties;
  focus: CSSObject;
};

export type SkeletonCSS = CSSObject & {
  backgroundColor: string;
  opacity: number;
};

// This is the shape of a theme 'mode', e.g. light, dark, or custom
export type Mode = {
  create: CreateCSS;
  iconButton: IconButtonCSS;
  navigation: NavigationCSS;
  primaryButton: PrimaryButtonCSS;
  productHome: ProductHomeCSS;
  search: SearchCSS;
  skeleton: SkeletonCSS;
};

export type NavigationTheme = {
  mode: Mode;
};

export type Colors = {
  backgroundColor: string;
  color: string;
};

export type GenerateThemeArgs = {
  name?: string;
  backgroundColor: string;
  highlightColor: string;
};
