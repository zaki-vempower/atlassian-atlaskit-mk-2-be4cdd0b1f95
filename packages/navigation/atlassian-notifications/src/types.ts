import { DetailedHTMLProps, IframeHTMLAttributes } from 'react';

type IframeProps = DetailedHTMLProps<
  IframeHTMLAttributes<HTMLIFrameElement>,
  HTMLIFrameElement
>;

export type NotificationsProps = Omit<IframeProps, 'src'> & {
  // Reserved for testing, avoid using this
  _url?: string;
  locale?: string;
  product?: 'confluence' | 'jira' | string;
  testId?: string;
};
