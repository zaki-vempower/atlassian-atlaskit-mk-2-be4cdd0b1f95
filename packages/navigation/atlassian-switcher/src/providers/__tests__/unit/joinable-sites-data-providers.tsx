import * as React from 'react';
import { shallow } from 'enzyme';

import {
  JoinableSitesProvider,
  prefetchJoinableSites,
} from '../../joinable-sites-data-provider';

import { JoinableSitesResponse } from '../../../types';

import { createProviderWithCustomFetchData } from '../../create-data-provider';

describe('joinable-sites-data-providers', () => {
  test('should render using the default provider', () => {
    const wrapper = shallow(
      <JoinableSitesProvider>{items => items}</JoinableSitesProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render using a custom provider', () => {
    const fetchData = () => Promise.resolve({ sites: [] });
    const customProvider = createProviderWithCustomFetchData<
      JoinableSitesResponse
    >('my-joinble-sites-provider', fetchData);
    const wrapper = shallow(
      <JoinableSitesProvider joinableSitesDataProvider={customProvider}>
        {items => items}
      </JoinableSitesProvider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should prefetch using the custom provider if passed down', () => {
    const fetchMethod = jest.fn();

    prefetchJoinableSites({
      fetchMethod,
    } as any);

    expect(fetchMethod).toBeCalledTimes(1);
  });
});
