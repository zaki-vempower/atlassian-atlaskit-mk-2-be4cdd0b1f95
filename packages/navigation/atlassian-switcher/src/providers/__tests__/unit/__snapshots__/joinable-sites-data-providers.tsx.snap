// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinable-sites-data-providers should render using a custom provider 1`] = `
<WithAnalyticsEvents(DataProvider(my-joinble-sites-provider))>
  <Component />
</WithAnalyticsEvents(DataProvider(my-joinble-sites-provider))>
`;

exports[`joinable-sites-data-providers should render using the default provider 1`] = `
<WithAnalyticsEvents(DataProvider(joinableSites))>
  <Component />
</WithAnalyticsEvents(DataProvider(joinableSites))>
`;
